pkg/operator2/oauth.go

```diff
@@ -83,7 +84,9 @@ func (c *authOperator) handleOAuthConfig(
 			},
 		)
 	}
-	handleDegraded(operatorConfig, "IdentityProviderConfig", v1helpers.NewMultiLineAggregate(errsIDP))
+	if len(errsIDP) > 0 {
```

Comment:
move the err != nil check back in the `degraded` method, possibly keep the `handleDegraded` name as that might be more appropriate due to that err handling, but I'll leave that up to you

Reply:
I externalized the error check from `degraded` because 4 out of 6 calls need to handle the error:

- https://github.com/marun/cluster-authentication-operator/blob/fix-ingress-state-flapping/pkg/operator2/operator.go#L252
- https://github.com/marun/cluster-authentication-operator/blob/fix-ingress-state-flapping/pkg/operator2/operator.go#L398
- https://github.com/marun/cluster-authentication-operator/blob/fix-ingress-state-flapping/pkg/operator2/operator.go#L408
- https://github.com/marun/cluster-authentication-operator/blob/fix-ingress-state-flapping/pkg/operator2/operator.go#L418

Why is it preferable to duplicate the error checking for a minority of cases?